helm/.dagger/main.go

```diff
@@ -128,55 +140,36 @@ func (h *Helm) SetVersion(
 	return updatedChartYaml, nil
 }
 
+// Test the packaging of the chart, without actually publishing it
+func (h *Helm) TestPublish(ctx context.Context) error {
+	_, err := h.chart().WithExec([]string{"helm", "package", "."}).Sync(ctx)
+	return err
+}
+
 // Package & publish chart to our registry
 func (h *Helm) Publish(
 	ctx context.Context,
 	tag string,
-
-	// +optional
-	// +default="https://github.com/dagger/dagger.git"
-	gitRepoSource string,
 	// +optional
 	githubToken *dagger.Secret,
-
-	// +optional
-	dryRun bool,
 ) error {
-	version := strings.TrimPrefix(tag, "helm/chart/")
-
-	helm := h.chart()
-	if githubToken != nil {
-		helm = helm.WithSecretVariable("GITHUB_TOKEN", githubToken)
-	}
-
-	if dryRun {
-		helm = helm.With(util.ShellCmd("helm package ."))
-	} else {
-		helm = helm.
-			With(util.ShellCmds(
+	_, err := h.chart().
+		With(func(c *dagger.Container) *dagger.Container {
+			if githubToken != nil {
+				return c.WithSecretVariable("GITHUB_TOKEN", githubToken)
+			}
+			return c
+		}).
+		With(func(c *dagger.Container) *dagger.Container {
+			version := strings.TrimPrefix(tag, "helm/chart/v")
+			script := strings.Join([]string{
 				"helm registry login ghcr.io/dagger --username dagger --password $GITHUB_TOKEN",
 				"helm package .",
-				"helm push dagger-helm-"+strings.TrimPrefix(version, "v")+".tgz oci://ghcr.io/dagger",
+				"helm push dagger-helm-" + version + ".tgz oci://ghcr.io/dagger",
 				"helm registry logout ghcr.io/dagger",
-			))
-	}
-	_, err := helm.Sync(ctx)
-	if err != nil {
-		return err
-	}
-
-	if semver.IsValid(version) {
-		if err := githubRelease(ctx, h.Dagger.Git, githubReleaseOpts{
-			tag:         "helm/chart/" + version,
-			target:      tag,
-			notes:       changeNotes(h.Dagger.Src, "helm/dagger", version),
-			gitRepo:     gitRepoSource,
-			githubToken: githubToken,
-			dryRun:      dryRun,
-		}); err != nil {
-			return err
-		}
-	}
```

Comment:
Wait, why is this removed? :thinking: This is a critical part of the release process for this component, we need to do this, or the tag+release will not actually be created. This is blocking, sorry!

I'm happy to work on this a bit :tada:, since I'm doing release things right now, so some options:
- We move `githubRelease` into a common shared module
- Or, each component that gets published isn't responsible for doing this bit - instead, this would get done from a top-level `release` module, that would call each individual components `Publish` function.
	- I *think* this would be my preference.
- Or, we just unblock temporarily, and copy-paste the code into here as well.

This is a good component to think about how we want the releasing structure of our modules to look like, since it's relatively much more self contained, so I think worth thinking about it a bit more.

Reply:
Probably a mistake when copying files over. Probably that snippet predates my original PR.

Reply:
Damnit, this pulls an additional dependency from the spaghetti, let me see if I can chop it off cleanly.